src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstier/service/ProtectLevelCalculator.kt

```diff
@@ -37,17 +36,19 @@ class ProtectLevelCalculator(
 
   fun calculateProtectLevel(
     crn: String,
-    offenderAssessment: OffenderAssessment?,
-    rsrScore: BigDecimal,
-    registrations: Collection<Registration>,
-    convictions: Collection<Conviction>
+    offenderAssessment: OffenderAssessment? = null,
+    rsrScore: BigDecimal = BigDecimal.ZERO,
+    rosh: Rosh? = null,
+    mappa: Mappa? = null,
+    complexityFactors: Collection<ComplexityFactor> = listOf(),
+    convictions: Collection<Conviction> = listOf()
```

Comment:
This I think is also always going to be an empty list so should not be defaulted

https://github.com/ministryofjustice/hmpps-tier/blob/main/src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstier/service/CommunityApiService.kt#L17

Reply:
I agree, its finding the right balance between these and having loads of nonsense in the tests. I wonder if Kotlin can help.